src/hotspot/cpu/riscv/riscv_v.ad

```diff
@@ -785,6 +782,120 @@ instruct vnegD(vReg dst, vReg src) %{
   ins_pipe(pipe_slow);
 %}
 
+// vector and reduction
+
+instruct reduce_andI(iRegINoSp dst, iRegIorL2I src1, vReg src2, vReg tmp) %{
+  predicate(n->in(2)->bottom_type()->is_vect()->element_basic_type() == T_INT);
```

Comment:
Does `Matcher::vector_element_basic_type(n->in(2)) == T_INT` work here?

Reply:
Thanks for the review, this encoding method does look a lot clearer, but there are many codes in riscv_ad that use the current writing method. Next, I will test this new encoding method. If everyone agrees, then Make unified changes.

Reply:
@theRealELiu Thank you for your suggestion, the current new nodes have been modified, and a new PR will be submitted in the future to modify other related places